src/main/java/com/getindata/kafka/connect/iceberg/sink/IcebergChangeEvent.java

```diff
@@ -358,14 +371,28 @@ private List<Types.NestedField> icebergSchema(JsonNode eventSchema, String schem
             if (items != null && items.has("type")) {
               String listItemType = items.get("type").textValue();
 
-              if (listItemType.equals("struct") || listItemType.equals("array") || listItemType.equals("map")) {
-                throw new RuntimeException("Complex nested array types are not supported," +
+
+              if (listItemType.equals("array") || listItemType.equals("map")) {
+                throw new RuntimeException("'array' and 'map' nested array types are not supported," +
                                            " array[" + listItemType + "], field " + fieldName);
               }
+              else {
+                if (listItemType.equals("struct")) {
+                  List<Types.NestedField> subSchema = icebergSchema(items, fieldName, columnId+2);
+                  schemaColumns.add(Types.NestedField.optional(columnId,
+                                                               fieldName,
+                                                               Types.ListType.ofOptional(columnId+1,
+                                                                                         Types.StructType.of(subSchema)),
+                                                               ""));
+                  columnId += subSchema.size() + 2;
+                }
+                else {
               // primitive coercions are not supported for list types, pass '""' for fieldTypeName
-              Type.PrimitiveType item = icebergFieldType(listItemType, "");
-              schemaColumns.add(Types.NestedField.optional(
-                  columnId, fieldName, Types.ListType.ofOptional(++columnId, item), ""));
+                  Type.PrimitiveType item = icebergFieldType(listItemType, "");
+                  schemaColumns.add(Types.NestedField.optional(
+                      columnId, fieldName, Types.ListType.ofOptional(++columnId, item), ""));
+                }
+              }
```

Comment:
Could you add a comment for the `columnId` manipulations? I'm not readily following that these do.

Reply:
Note the adjustments to columnID for the bare Struct type:  https://github.com/reifyhealth/kafka-connect-iceberg-sink/blob/93eedacda7f2bc37a04df45168640f4a0d808693/src/main/java/com/getindata/kafka/connect/iceberg/sink/IcebergChangeEvent.java#L376

Reply:
In the case of `Array<Struct>` an additional columnId offset needs to be added for the surrounding `Array` typed  container, hence the additional increments to columnId.

Reply:
Got it, thanks